components/raftstore/src/coprocessor/split_check/mod.rs

```diff
@@ -128,3 +128,15 @@ impl<'a, E> Host<'a, E> {
         self.cfg.region_bucket_size.0
     }
 }
+
+#[inline]
+pub fn calc_split_keys_count(count_per_region: u64, split_threshold: u64, batch_split_limit: u64) -> u64 {
+    let actual_split_limit = if count_per_region % split_threshold == 0 {
```

Comment:
I think it's `cmp::min(count_per_region.saturating_sub(1) / split_threshold, batch_split_limit)`.

Reply:
good catch on potential overflow.  But the "if block" is still needed

Reply:
Why?

Reply:
ok. used the suggestion below.